### PR TITLE
test: faster node checkpointing test (2m to 7s)

### PR DESCRIPTION
**Pull request by snormore**
The `node_checkpointing` test currently takes ~2m to complete. The test builds a checkpoint and runs through 3 iterations of: load checkpoint into the node, start the node, wait for it to be ready, then shutdown the node.

This PR speeds the test up to about 7s, doing 10 iterations of the same process. It does that by mostly replacing the 30s sleeps with coordination via `wait_for_ready`.

### CODE REVIEW

File: core/application/src/app.rs [hunk 1]
```diff
@@ -100,32 +99,16 @@ impl<C: NodeComponents> ApplicationInterface<C> for Application<C> {
         checkpoint: Vec<u8>,
         checkpoint_hash: [u8; 32],
     ) -> Result<()> {
-        // Due to a race condition on shutdowns when a node checkpoints, we should sleep and try
```
Comment (snormore) [inline core/application/src/app.rs:103]: I wasn't able to reproduce this anymore after running the test in a loop non-stop for a while. I was able to reproduce the rocks "`LOCK` still held by current process" error from here if there is no wait_for_ready or sleep after startup before shutdown, but this loop doesn't solve that, it just loops and fails at the end.

**Review by ozwaldorf — APPROVED** [review]
lgtm, nice :+1:

File: core/application/src/app.rs (hunk 1 above)
Comment (ozwaldorf) [reply core/application/src/app.rs:103]: but we call wait for ready which solves it right?

**Comment by snormore** [comment]
Made some peripheral changes to this since last review. Probably worth another look 🙏

**Review by ozwaldorf — APPROVED** [review]

**Pull request merged**